sdk/dynamicconfig/sdk/metric/dynamicextension.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"sync"
+
+	pb "github.com/vmingchen/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+
+	"go.opentelemetry.io/contrib/sdk/dynamicconfig/internal/metricpattern"
+)
+
+// This file contains extraneous functionality needed to allow per-metric configuration
+// for the Accumulator.
+
+// Extension to accumulator which allows per-metric collection.
```

Comment:
What do you think about renaming it something more specific like "PeriodMatcher," since it's function seems to be to match names to periods? It may be a little difficult to grasp what a "DynamicExtension" struct does, but more immediately obvious what a "PeriodMatcher" should do?

Reply:
With the changes you and Ming and I discussed, `DynamicExtension` is now the sole access point for the current list of schedules, so "PeriodMatcher" might not be the best descriptor of it's functionality. However, I definitely agree that "DynamicExtension" is a clunky name, and would love ideas about alternate ones!